### DISCUSSION TITLE
DHT20 and Tasmota on a 8266 Wemos D1

### DISCUSSION THREAD

**Discussion by onki69** [arendst/Tasmota#22210] [Q&A]
Hi I am struggling to create a temperature sensor based on the fairly new DHT20.
It seems to be an updated version of the DHT11 but more precise.
How do I use this sensor on my tasmota flashed Wemos D1 board.
I know that I need power supply (3V3) and GND. Buit do I need to set the GPIOs to DHT or do I need to specify SDA and SCL?

**Comment by sfromis** [comment]
If it was compatible with the pretty bad DHT11, you should configure like that thing, but it looks like it is completely unrelated, not even similar or "updated" internals.

Instead it is is likely an AHT20 on I2C, and needing a build with `USE_AHT2x`

**Comment by onki69** [comment]
OK - I changed to a DHT22 and now it works. I can see temperature and humidity in the main screen.

Is there any possibitlity to use this temperature sensor together with a tasmota relay as a thermostat?
The power socket should switch off if a maximum temperature is reached.

**Reply by sfromis** [reply]
Great if it worked for you that way, but that does not even look equivalent to published information about DHT20, which says that it is an I2C-based sensor, which the DHT20 (or DHT10) is not.

When it comes to a thermostat function, yes, Tasmota does include a Thermostat feature, or you can do it with simple rules.

**Comment by onki69** [comment]
Thanks. I thought that using a rule is a possible way.
Is there a code example how to create such a rule?
I guess the rule must be defined on the relay (Shelly Plus 1PM with Tasmota).
How do I refer to the external Tasmota temp sensor?

**Reply by sfromis** [reply]
https://tasmota.github.io/docs/Rules/#simple-thermostat-example
